cli/src/processSourceFiles.ts

```diff
@@ -57,79 +99,126 @@ export const processSourceFiles = async ({
     process.exit(1); // Exit with error code
   }
 
+  // Validate that path is a directory
+  const stats = fs.statSync(buildPath);
+  if (!stats.isDirectory()) {
+    console.error(`Path must be a directory: ${buildPath}`);
+    process.exit(1); // Exit with error code
+  }
+
   try {
-    // load files content
-    let jsContent = fs.readFileSync(jsFilePath, fileEncoding);
-    let mapContent = fs.readFileSync(mapFilePath, fileEncoding);
-
-    let newJsContent: string;
-    let newMapContent: string;
-
-    // if an app version is provided, inject it into the source code
-    // note that this is not mandatory, as the app version cam also be provided during sdk initialization.
-    // If neither is provided, the SDK will report the app version as "EmbIOAppVersionX.X.X"
-    if (appVersion) {
-      // for that, generate a 20 chars long appVersion by adding leading spaces to the appVersion
-      // if it is less than 20 chars long
-      const appVersionLength = appVersion.length;
-      if (appVersionLength < 20) {
-        appVersion = appVersion.padStart(20, ' ');
+    // Iterate over directory to find .js files with corresponding .js.map files
+    const files = fs.readdirSync(buildPath);
+    const jsFiles = files.filter(file => file.endsWith('.js'));
+
+    console.log(
+      `Found ${jsFiles.length} JavaScript files in directory: ${buildPath}`
+    );
+
+    let appVersionReplaced = false;
+    for (const jsFile of jsFiles) {
```

Comment:
@embrace-io/team-web Something I forgot to ask, should I do recursive search here? Is it normal for build directories to have js files in different subdirs? Or maybe allow doing it configured with a different param?